lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -237,8 +237,16 @@ tokenize([$~, S, H | _] = Original, Line, Column, _Scope, Tokens) when ?is_upcas
 
 tokenize([$?, $\\, H | T], Line, Column, Scope, Tokens) ->
   Char = elixir_interpolation:unescape_map(H),
+  NewScope = if
+    H =:= Char andalso (?is_upcase(H) orelse ?is_downcase(H)) ->
```

Comment:
Could this be rewritten to `(H =:= Char) and (H =/= $\\)`? This is the only case where `\\` is needed to return itself, right?

Reply:
I actually originally had this check, but there was a few warnings showing up in `ExUnit`, because it does `?\"` a couple of times. While that escape isn't necessary there, it is a common escape in other contexts, so I'm not sure whether you'd prefer printing warnings in that case or not.

Do you have any thoughts about that?

Reply:
I would say let’s try to warn for all and if the push back is big, we can relax it a bit. WDYT?

Reply:
That sounds good to me! I'll fix those warnings as part of this PR.